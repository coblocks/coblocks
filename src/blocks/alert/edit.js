/**
 * External dependencies
 */
import classnames from 'classnames';

/**
 * Internal dependencies
 */
import Controls from './controls';
import Inspector from './inspector';
import applyWithColors from './colors';

/**
 * WordPress dependencies
 */
import { _x } from '@wordpress/i18n';
import { Component, Fragment } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { RichText } from '@wordpress/block-editor';

/**
 * Block edit function
 */
class Edit extends Component {
	componentDidMount() {
		const { attributes, setAttributes } = this.props;

		// Convert is-{type}-alert to is-style-{type}.
		// See: https://github.com/godaddy/coblocks/pull/781
		if ( /is-\w+-alert/.test( attributes.className ) ) {
			let newClassName = attributes.className;

			newClassName = newClassName.replace( 'is-default-alert', 'is-style-info' );
			newClassName = newClassName.replace( /is-(\w+)-alert/, 'is-style-$1' );
			setAttributes( { className: newClassName } );
		}
	}

	componentDidUpdate( prevProps ) {
		const { attributes, setAttributes } = this.props;

		// Reset color selections when a new style has been selected.
		// If the legacy alert class is detected, we want to retain the custom color selections.
		if ( ! /is-\w+-alert/.test( prevProps.attributes.className ) && prevProps.attributes.className !== attributes.className ) {
			setAttributes( { backgroundColor: '', customBackgroundColor: '', textColor: '', customTextColor: '' } );
		}
	}

	render() {
		const {
			attributes,
			backgroundColor,
			className,
			isSelected,
			setAttributes,
			textColor,
		} = this.props;

		const {
			textAlign,
			title,
			value,
		} = attributes;

		return (
			<Fragment>
				{ isSelected && (
					<Controls
						{ ...this.props }
					/>
				) }
				{ isSelected && (
					<Inspector
						{ ...this.props }
					/>
				) }
				<div
					className={ classnames(
						className, {
							'has-background': backgroundColor.color,
							'has-text-color': textColor.color,
							[ `has-text-align-${ textAlign }` ]: textAlign,
							[ backgroundColor.class ]: backgroundColor.class,
							[ textColor.class ]: textColor.class,
						}
					) }
					style={ {
						backgroundColor: backgroundColor.color,
						color: textColor.color,
					} }
				>
					{ ( ! RichText.isEmpty( title ) || isSelected ) && (
						<RichText
							placeholder={ _x( 'Write title...', 'Placeholder text for input box', 'coblocks' ) }
							value={ title }
							className="wp-block-coblocks-alert__title"
							onChange={ ( value ) => setAttributes( { title: value } ) }
							keepPlaceholderOnFocus
						/>
					) }
					<RichText
						placeholder={ _x( 'Write text...', 'Placeholder text for input box', 'coblocks' ) }
						value={ value }
						className="wp-block-coblocks-alert__text"
						onChange={ ( value ) => setAttributes( { value: value } ) }
						keepPlaceholderOnFocus
					/>
				</div>
			</Fragment>
		);
	}
}

export default compose( [
	applyWithColors,
] )( Edit );
