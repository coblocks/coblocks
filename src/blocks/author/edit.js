/**
 * External dependencies
 */
import classnames from 'classnames';

/**
 * Internal dependencies
 */
import Controls from './controls';
import Inspector from './inspector';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';
import { compose } from '@wordpress/compose';
import { Component, Fragment } from '@wordpress/element';
import { mediaUpload } from '@wordpress/editor';
import { RichText, InnerBlocks, MediaUpload, MediaUploadCheck, withColors, withFontSizes } from '@wordpress/block-editor';
import { Button, Dashicon, DropZone } from '@wordpress/components';

class AuthorEdit extends Component {
	constructor() {
		super( ...arguments );
		this.addImage = this.addImage.bind( this );
		this.onSelectImage = this.onSelectImage.bind( this );
	}

	onSelectImage( media ) {
		if ( media && media.url ) {
			this.props.setAttributes( { imgUrl: media.url } );
		}
	}

	addImage( files ) {
		mediaUpload( {
			allowedTypes: [ 'image' ],
			filesList: files,
			onFileChange: ( [ media ] ) => this.onSelectImage( media ),
		} );
	}

	render() {
		const {
			attributes,
			className,
			isSelected,
			setAttributes,
			backgroundColor,
			textColor,
			fontSize,
		} = this.props;

		const {
			biography,
			imgUrl,
			name,
		} = attributes;

		const dropZone = (
			<DropZone
				onFilesDrop={ this.addImage }
				label={ _x( 'Drop to upload as avatar', 'image to represent the post author', 'coblocks' ) }
			/>
		);

		const classes = classnames(
			className, {
				'has-background': backgroundColor.color,
				'has-text-color': textColor.color,
				[ backgroundColor.class ]: backgroundColor.class,
				[ textColor.class ]: textColor.class,
			}
		);

		const styles = {
			backgroundColor: backgroundColor.color,
			color: textColor.color,
			fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
		};

		const onUploadImage = ( media ) => setAttributes( { imgUrl: media.url, imgId: media.id } );

		return (
			<Fragment>
				{ isSelected && (
					<Controls
						{ ...this.props }
					/>
				) }
				{ isSelected && (
					<Inspector
						{ ...this.props }
					/>
				) }
				<div className={ classes } style={ styles }>
					{ dropZone }
					<figure className={ `${ className }__avatar` }>
						<MediaUploadCheck>
							<MediaUpload
								onSelect={ onUploadImage }
								allowedTypes={ [ 'image' ] }
								value={ imgUrl }
								render={ ( { open } ) => (
									<Button onClick={ open }>
										{ ! imgUrl ?
											<Dashicon icon="format-image" /> :
											<img className={ `${ className }__avatar-img` }
												src={ imgUrl }
												alt="avatar"
											/>
										}
									</Button>
								) }
							>
							</MediaUpload>
						</MediaUploadCheck>
					</figure>
					<div className={ `${ className }__content` }>
						<RichText
							identifier="name"
							multiline={ false }
							tagName="span"
							className={ `${ className }__name` }
							placeholder={
								// translators: placeholder text used for the author name
								__( 'Write author name…', 'coblocks' )
							}
							value={ name }
							onChange={ ( nextName ) => {
								setAttributes( { name: nextName } );
							} }
						/>
						<RichText
							identifier="biography"
							multiline={ false }
							tagName="p"
							className={ `${ className }__biography` }
							placeholder={
								// translators: placeholder text used for the biography
								__( 'Write a biography that distills objective credibility and authority to your readers…', 'coblocks' )
							}
							value={ biography }
							onChange={ ( nextBio ) => {
								setAttributes( { biography: nextBio } );
							} }
						/>
						<InnerBlocks
							template={ [ [ 'core/button', { placeholder: _x( 'Author link…', 'content placeholder', 'coblocks' ) } ] ] }
							templateLock="all"
							allowedBlocks={ [ 'core/button' ] }
							templateInsertUpdatesSelection={ false }
						/>
					</div>
				</div>
			</Fragment>
		);
	}
}

export default compose( [
	withColors( 'backgroundColor', { textColor: 'color' } ),
	withFontSizes( 'fontSize' ),
] )( AuthorEdit );
