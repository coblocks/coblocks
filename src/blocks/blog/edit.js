/**
 * Internal dependencies
 */
import './styles/editor.scss';
import './styles/style.scss';
import Slider from 'react-slick';

/**
 * External dependencies
 */
import { isUndefined, pickBy } from 'lodash';
import classnames from 'classnames';

/**
 * WordPress dependencies
 */
import { Component, RawHTML } from '@wordpress/element';
import {
	PanelBody,
	Placeholder,
	QueryControls,
	RangeControl,
	Spinner,
	ToggleControl,
	Toolbar,
	RadioControl,
	TextControl,
	Button,
	ServerSideRender,
} from '@wordpress/components';
import apiFetch from '@wordpress/api-fetch';
import { addQueryArgs } from '@wordpress/url';
import { __ } from '@wordpress/i18n';
import { dateI18n, format, __experimentalGetSettings } from '@wordpress/date';
import { withSelect } from '@wordpress/data';
import blogIcons from './icons';
import includes from 'lodash/includes';
import { Fragment } from 'react';
import SlickSliderPanel from '../../components/slick-slider-panel';

/**
 * Module Constants
 */
const { InspectorControls, BlockControls, PlainText } = wp.editor;

const CATEGORIES_LIST_QUERY = {
	per_page: -1,
};
const MAX_POSTS_COLUMNS = 3;

class LatestPostsEdit extends Component {
	constructor() {
		super( ...arguments );
		this.state = {
			categoriesList: [],
			editing: ! this.props.attributes.externalRssUrl,
		};

		this.onSubmitURL = this.onSubmitURL.bind( this );
	}

	componentDidMount() {
		this.isStillMounted = true;
		this.fetchRequest = apiFetch( {
			path: addQueryArgs( '/wp-json/wp/v2/categories', CATEGORIES_LIST_QUERY ),
		} ).then(
			( categoriesList ) => {
				if ( this.isStillMounted ) {
					this.setState( { categoriesList } );
				}
			}
		).catch(
			( e ) => {
				if ( this.isStillMounted ) {
					this.setState( { categoriesList: [] } );
				}
			}
		);
	}

	onSubmitURL( event ) {
		event.preventDefault();

		const { externalRssUrl } = this.props.attributes;
		if ( externalRssUrl ) {
			this.setState( { editing: false } );
		}
	}

	componentWillUnmount() {
		this.isStillMounted = false;
	}

	render() {
		const { attributes, setAttributes, className, latestPosts } = this.props;

		const isListStyle = includes( className, 'is-style-list' );
		const isGridStyle = includes( className, 'is-style-grid' );
		const isCarouselStyle = includes( className, 'is-style-carousel' );

		const { categoriesList, editing } = this.state;
		const {
			displayPostContent,
			displayPostDate,
			displayPostLink,
			displayCategories,
			postLink,
			postFeedType,
			externalRssUrl,
			columns,
			order,
			orderBy,
			categories,
			postsToShow,
			excerptLength,
			listPosition,
			prevNextButtons,
			visibleItems,
			draggable,
			autoPlay,
			autoPlaySpeed } = attributes;

		const postSettingsControls = (
			<PanelBody title={ __( 'Post Settings' ) }>
				<ToggleControl
					label={ __( 'Display Date' ) }
					checked={ displayPostDate }
					help={ __( 'Showing the publish date.' ) }
					onChange={ ( value ) => setAttributes( { displayPostDate: value } ) }
				/>
				<ToggleControl
					label={ __( 'Display Link' ) }
					checked={ displayPostLink }
					help={ __( 'Showing links to individual posts.' ) }
					onChange={ ( value ) => setAttributes( { displayPostLink: value } ) }
				/>
				<ToggleControl
					label={ __( 'Display Excerpt' ) }
					checked={ displayPostContent }
					help={ __( 'Showing the post excerpt.' ) }
					onChange={ ( value ) => setAttributes( { displayPostContent: value } ) }
				/>
				<ToggleControl
					label={ __( 'Display Categories' ) }
					checked={ displayCategories }
					help={ __( 'Showing posts categories.' ) }
					onChange={ ( value ) => setAttributes( { displayCategories: value } ) }
				/>
				{ displayPostContent &&
					<RangeControl
						label={ __( 'Max words in post excerpt' ) }
						value={ excerptLength }
						onChange={ ( value ) => setAttributes( { excerptLength: value } ) }
						min={ 10 }
						max={ 100 }
					/>
				}
			</PanelBody>
		);

		const sortingAndFiltering = (
			<PanelBody title={ __( 'Sorting and Filtering' ) }>
				<QueryControls
					{ ...{ order, orderBy } }
					numberOfItems={ postsToShow }
					categoriesList={ categoriesList }
					selectedCategoryId={ categories }
					onOrderChange={ ( value ) => setAttributes( { order: value } ) }
					onOrderByChange={ ( value ) => setAttributes( { orderBy: value } ) }
					onCategoryChange={ ( value ) => setAttributes( { categories: '' !== value ? value : undefined } ) }
					onNumberOfItemsChange={ ( value ) => setAttributes( { postsToShow: value } ) }
				/>
				{ isGridStyle &&
				<RangeControl
					label={ __( 'Columns' ) }
					value={ columns }
					onChange={ ( value ) => setAttributes( { columns: value } ) }
					min={ 2 }
					max={ 4 }
					required
				/>
				}
			</PanelBody>
		);

		const feedType = (
			<PanelBody title={ __( 'Feed type' ) }>
				<RadioControl
					label={ __( 'Post Feed' ) }
					selected={ postFeedType }
					options={ [
						{ label: 'My Blog', value: 'internal' },
						{ label: 'External Blog Feed', value: 'external' },
					] }
					onChange={ ( value ) => setAttributes( { postFeedType: value } ) }
				/>
				{ postFeedType === 'external' &&
				<div>
					<Button isLarge onClick={ () => this.setState( { editing: true } ) } >
						{ 'Edit external RSS' }
					</Button>
				</div>
				}
			</PanelBody>
		);

		const hasPosts = Array.isArray( latestPosts ) && latestPosts.length;
		if ( ! hasPosts && postFeedType === 'internal' ) {
			return (
				<Fragment>
					<InspectorControls>
						{ feedType }
					</InspectorControls>
					<Placeholder
						icon="admin-post"
						label={ __( 'Blog' ) }
					>
						{ ! Array.isArray( latestPosts ) ?
							<Spinner /> :
							__( 'No posts found.' )
						}
					</Placeholder>
				</Fragment>
			);
		}

		const displayPosts = latestPosts.length > postsToShow ?
			latestPosts.slice( 0, postsToShow ) :
			latestPosts;

		const toolbarControls = [ {
			icon: blogIcons.mediaCardRight,
			title: __( 'Image on left' ),
			isActive: listPosition === 'left',
			onClick: () => setAttributes( { listPosition: 'left' } ),
		}, {
			icon: blogIcons.mediaCardLeft,
			title: __( 'Image on right' ),
			isActive: listPosition === 'right',
			onClick: () => setAttributes( { listPosition: 'right' } ),
		} ];

		const slickSettings = {
			autoPlay: autoPlay,
			autoPlaySpeed: autoPlaySpeed,
			dots: false,
			arrows: prevNextButtons,
			infinite: true,
			draggable: draggable,
			adaptiveHeight: false,
			speed: 500,
			slidesToShow: visibleItems,
			slidesToScroll: 1,
		};

		const dateFormat = __experimentalGetSettings().formats.date;

		if ( this.state.editing && postFeedType === 'external' ) {
			return (
				<Fragment>
					<InspectorControls>
						{ feedType }
					</InspectorControls>
					<Placeholder
						icon="rss"
						label="RSS"
					>
						<form onSubmit={ this.onSubmitURL }>
							<TextControl
								placeholder={ __( 'Enter URL hereâ€¦' ) }
								value={ externalRssUrl }
								onChange={ ( value ) => setAttributes( { externalRssUrl: value } ) }
								className={ 'components-placeholder__input' }
							/>
							<Button isLarge type="submit">
								{ __( 'Use URL' ) }
							</Button>
						</form>
					</Placeholder>
				</Fragment>
			);
		}

		return (
			<div>
				<InspectorControls>
					{ feedType }
					{ postSettingsControls }
					{ isCarouselStyle &&
						<SlickSliderPanel { ...this.props } />
					}
					{ sortingAndFiltering }
				</InspectorControls>
				<BlockControls>
					{ isListStyle &&
						<Toolbar
							controls={ toolbarControls }
						/>
					}
				</BlockControls>
				{ postFeedType === 'external' &&
					<ServerSideRender
						block="coblocks/blog"
						attributes={ this.props.attributes }
						className="coblocks-slick"
					/>
				}
				{ postFeedType === 'internal' && ! isCarouselStyle &&
				<ul
					className={ classnames( this.props.className, {
						'image-to-left': listPosition === 'left',
						'image-to-right': listPosition === 'right',
						'wp-block-coblocks-blog__list': true,
						'has-dates': displayPostDate,
						[ `columns-${ columns }` ]: isGridStyle,
					} ) }
				>
					{ displayPosts.map( ( post, i ) => {
						const featuredImageUrl = post.featured_media_object ? post.featured_media_object.source_url : null;
						const featuredImageStyle = 'url(' + featuredImageUrl + ')';
						const titleTrimmed = post.title.rendered.trim();
						let excerpt = post.excerpt.rendered;
						if ( post.excerpt.raw === '' ) {
							excerpt = post.content.raw;
						}
						const excerptElement = document.createElement( 'div' );
						excerptElement.innerHTML = excerpt;
						excerpt = excerptElement.textContent || excerptElement.innerText || '';
						return (
							<li key={ i }>
								{ featuredImageUrl &&
								<div className="wp-block-coblocks-blog__post-image-wrapper"
									 style={ { backgroundImage: featuredImageStyle } }></div>
								}
								<div className="wp-block-coblocks-blog__post-info">
									{ displayPostDate && post.date_gmt &&
									<time dateTime={ format( 'c', post.date_gmt ) }
										  className="wp-block-coblocks-blog__post-date">
										{ dateI18n( dateFormat, post.date_gmt ) }
									</time>
									}
									<h5>
										{ titleTrimmed ? (
											<RawHTML>
												{ titleTrimmed }
											</RawHTML>
										) :
											__( '(Untitled)' )
										}
									</h5>
									{ displayPostContent &&
									<div className="wp-block-coblocks-blog__post-excerpt">
										<p>
											<RawHTML
												key="html"
											>
												{ excerpt.trim().split( ' ', excerptLength ).join( ' ' ) }
											</RawHTML>
										</p>
									</div>
									}
									{ displayPostLink &&
									<PlainText
										className="wp-block-coblocks-blog__post-read-more"
										onChange={ ( newPostLink ) => setAttributes( { postLink: newPostLink } ) }
										value={ postLink }
										placeholder={ __( 'Continue Reading' ) }
									/>
									}
								</div>
							</li>
						);
					} ) }
				</ul>
				}
				{ postFeedType === 'internal' && isCarouselStyle &&
				<Slider { ...slickSettings } className={ classnames( this.props.className ) }>
					{ displayPosts.map( ( post, i ) => {
						const featuredImageUrl = post.featured_media_object ? post.featured_media_object.source_url : null;
						const featuredImageStyle = 'url(' + featuredImageUrl + ')';
						const titleTrimmed = post.title.rendered.trim();
						let excerpt = post.excerpt.rendered;
						if ( post.excerpt.raw === '' ) {
							excerpt = post.content.raw;
						}
						const excerptElement = document.createElement( 'div' );
						excerptElement.innerHTML = excerpt;
						excerpt = excerptElement.textContent || excerptElement.innerText || '';
						return (
							<div className="coblocks-blog-post--item" key={ i }>
								<div className="coblocks-blog-post--item-inner">
									{ featuredImageUrl &&
									<div className="wp-block-coblocks-blog__post-image-wrapper"
										 style={ { backgroundImage: featuredImageStyle } }></div>
									}
									<div className={ classnames('wp-block-coblocks-blog__post-info', {
										'full-height': !featuredImageUrl,
									} ) }>
										{ displayPostDate && post.date_gmt &&
										<time dateTime={ format( 'c', post.date_gmt ) }
											  className="wp-block-coblocks-blog__post-date">
											{ dateI18n( dateFormat, post.date_gmt ) }
										</time>
										}
										<h5>
											{ titleTrimmed ? (
												<RawHTML>
													{ titleTrimmed }
												</RawHTML>
											) :
												__( '(Untitled)' )
											}
										</h5>
										{ displayPostContent &&
										<div className="wp-block-coblocks-blog__post-excerpt">
											<p>
												<RawHTML
													key="html"
												>
													{ excerpt.trim().split( ' ', excerptLength ).join( ' ' ) }
												</RawHTML>
											</p>
										</div>
										}
										{ displayPostLink &&
										<PlainText
											className="wp-block-coblocks-blog__post-read-more"
											onChange={ ( newPostLink ) => setAttributes( { postLink: newPostLink } ) }
											value={ postLink }
											placeholder={ __( 'Continue Reading' ) }
										/>
										}
									</div>
								</div>
							</div>
						);
					} ) }
				</Slider>
				}
			</div>
		);
	}
}

export default withSelect( ( select, props ) => {
	const { postsToShow, order, orderBy, categories } = props.attributes;
	const { getEntityRecords } = select( 'core' );
	const latestPostsQuery = pickBy( {
		categories,
		order,
		orderby: orderBy,
		per_page: postsToShow,
	}, ( value ) => ! isUndefined( value ) );

	let latestPosts = getEntityRecords( 'postType', 'post', latestPostsQuery );
	if ( latestPosts ) {
		latestPosts = latestPosts.map( ( post ) => {
			return {
				...post,
				featured_media_object: select( 'core' ).getMedia( post.featured_media ),
			};
		} );
	}

	return {
		latestPosts: latestPosts,
	};
} )( LatestPostsEdit );
