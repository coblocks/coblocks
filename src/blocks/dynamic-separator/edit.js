/**
 * External dependencies
 */
import classnames from 'classnames';

/**
 * Internal dependencies
 */
import Inspector from './inspector';
import applyWithColors from './colors';

/**
 * WordPress dependencies
 */
import { Component, Fragment } from '@wordpress/element';
import { compose } from '@wordpress/compose';
import { ResizableBox } from '@wordpress/components';

/**
 * Block edit function
 */
class DynamicSeparatorEdit extends Component {
	render() {
		const {
			attributes,
			className,
			isSelected,
			setAttributes,
			color,
		} = this.props;

		const { height } = attributes;

		const hrClasses = classnames( className, 'wp-block-separator', {
			'is-selected': isSelected,
			'has-text-color': color.color,
			[ color.class ]: color.class,
			'is-style-dots': ! className.includes( 'is-style-line' ) && ! className.includes( 'is-style-fullwidth' ) && ! className.includes( 'is-style-dots' ),
		} );

		return (
			<Fragment>
				{ isSelected && <Inspector { ...this.props } /> }
				<ResizableBox
					className={ hrClasses }
					style={ {
						color: color.color,
					} }
					size={ {
						height,
					} }
					minHeight="20"
					minWidth="100%"
					enable={ {
						top: false,
						right: false,
						bottom: true,
						left: false,
						topRight: false,
						bottomRight: false,
						bottomLeft: false,
						topLeft: false,
					} }
					onResizeStop={ ( _event, _direction, _elt, delta ) => {
						const spacerHeight = parseInt( height + delta.height, 10 );
						setAttributes( {
							height: spacerHeight,
						} );
					} }
				/>
			</Fragment>
		);
	}
}

export default compose( [ applyWithColors ] )( DynamicSeparatorEdit );
