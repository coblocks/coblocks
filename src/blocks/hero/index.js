/**
 * Styles.
 */
import './styles/editor.scss';
import './styles/style.scss';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import CSSGridAttributes from '../../components/grid-control/attributes';
import DimensionsAttributes from '../../components/dimensions-control/attributes';
import ResponsiveBaseControlAttributes from '../../components/responsive-base-control/attributes';
import { BackgroundAttributes } from '../../components/background';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...CSSGridAttributes,
	...DimensionsAttributes,
	...BackgroundAttributes,
	...ResponsiveBaseControlAttributes,
	...metadata.attributes,
};

const settings = {
	title: _x( 'Hero', 'block name', 'coblocks' ),
	description: __( 'An introductory area of a page accompanied by a small amount of text and a call to action.', 'coblocks' ),
	icon,
	keywords: [ _x( 'button', 'block keyword', 'coblocks' ), _x( 'cta', 'block keyword', 'coblocks' ), _x( 'call to action', 'block keyword', 'coblocks' ) ],
	supports: {
		align: [ 'wide', 'full' ],
		coBlocksSpacing: true,
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings, attributes };
