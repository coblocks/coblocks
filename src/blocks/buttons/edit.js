/**
 * External dependencies
 */
import classnames from 'classnames';
import memoize from 'memize';
import times from 'lodash/times';

/**
 * Internal dependencies
 */
import Inspector from './inspector';
import Controls from './controls';

/**
 * WordPress dependencies
 */
import { Component, Fragment } from '@wordpress/element';
import { InnerBlocks } from '@wordpress/block-editor';
import { createBlock } from '@wordpress/blocks';

/**
 * Constants
 */
const ALLOWED_BLOCKS = [ 'core/button' ];

/**
 * Returns the layouts configuration for a given number of feature items.
 *
 * @param {number} count Number of feature items.
 *
 * @return {Object[]} Columns layout configuration.
 */
const getCount = memoize( ( count ) => {
	return times( count, () => [ 'core/button' ] );
} );

class ButtonsEdit extends Component {
	render() {
		const {
			attributes,
			className,
			isSelected,
			clientId,
		} = this.props;

		const {
			items,
			contentAlign,
			isStackedOnMobile,
		} = attributes;

		const classes = classnames(
			'wp-block-coblocks-buttons__inner', {
				[ `flex-align-${ contentAlign }` ]: contentAlign,
				'is-stacked-on-mobile': isStackedOnMobile,
			}
		);

		if ( wp.data.select( 'core/block-editor' ).canInsertBlockType( 'core/buttons', clientId ) ) {
			const thisBlock = wp.data.select( 'core/block-editor' ).getBlocksByClientId( clientId )[ 0 ];
			const coreButtons = createBlock( 'core/buttons', { align: thisBlock.attributes.contentAlign }, thisBlock.innerBlocks );
			wp.data.dispatch( 'core/block-editor' ).replaceBlocks( clientId, coreButtons );
		}

		return (
			<Fragment>
				{ isSelected && (
					<Inspector
						{ ...this.props }
					/>
				) }
				{ isSelected && (
					<Controls
						{ ...this.props }
					/>
				) }
				<div className={ className }>
					<div className={ classes }>
						<InnerBlocks
							allowedBlocks={ ALLOWED_BLOCKS }
							template={ getCount( items ) }
							templateLock="all"
							templateInsertUpdatesSelection={ false }
						/>
					</div>
				</div>
			</Fragment>
		);
	}
}

export default ButtonsEdit;
