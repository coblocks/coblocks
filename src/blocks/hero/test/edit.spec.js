/**
 * External dependencies
 */
import ShallowRenderer from 'react-test-renderer/shallow';
import { getBlockDefaultClassName } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name } from '../index';
import { Edit } from '../edit';

const defaultProps = {
	className: getBlockDefaultClassName( name ),
	attributes: {},
};

describe( 'coblocks/hero edit', () => {
	it( 'should render', () => {
		const renderer = new ShallowRenderer();
		renderer.render( <Edit { ...defaultProps } /> );
		const component = renderer.getRenderOutput();

		expect( component ).toBeDefined();
		expect( component ).toMatchSnapshot();
	} );
} );
