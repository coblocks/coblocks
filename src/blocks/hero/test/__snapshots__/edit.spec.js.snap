// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/hero edit should render 1`] = `
<React.Fragment>
  <BackgroundDropZone
    attributes={Object {}}
    className="wp-block-coblocks-hero"
    label="Add backround to hero"
  />
  <div
    className="wp-block-coblocks-hero"
  >
    <ResizableBox
      className="wp-block-coblocks-hero__inner"
      enable={
        Object {
          "bottom": true,
          "bottomLeft": false,
          "bottomRight": false,
          "left": false,
          "right": false,
          "top": false,
          "topLeft": false,
          "topRight": false,
        }
      }
      minHeight="500"
      onResizeStop={[Function]}
      size={
        Object {
          "height": undefined,
        }
      }
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": undefined,
          "backgroundPosition": undefined,
          "color": undefined,
          "minHeight": undefined,
          "paddingBottom": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
        }
      }
    />
  </div>
</React.Fragment>
`;
